nodejs/awsx/apigateway/cognitoAuthorizer.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These APIs are currently experimental and may change.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+/**
+ * CognitoAuthorizer provides the definition for a Cognito User Pools Authorizer for API Gateway.
+ */
+export interface CognitoAuthorizer {
+    /**
+     * The name for the Authorizer to be referenced as. This must be unique for each unique
+     * authorizer within the API. If no name if specified, a name will be generated for you.
+     */
+    authorizerName?: string;
+
+    /**
+     * parameterName is the name of the header containing the authorization token.
+     */
+    parameterName: string;
+
+    /**
+     * The ARNs of the Cognito User Pools to use.
+     */
+    providerARNs: Array<pulumi.Input<string> | aws.cognito.UserPool>;
```

Comment:
fyi: `(pulumi.Input<string> | aws.cognito.UserPool)[]` is also a way to write this.  Up to you which you prefer.